### ISSUE TITLE
this.send operates referentially on the command object. 

### ISSUE THREAD

**Issue by tjgionet** [IntrospectiveSystems/xGraph#92]
```
let temp = {Cmd:"Command"};
let promiseArray = [new promise(res=>this.send(temp, DestA, res)), new promise(res=>this.send(temp, DestB, res))];
await promise.all(promiseArray);
```

DestA may not receive the command, as the internal destination may be changed (by the second send function) before sending is complete

**Comment by valyrie97** [comment]
we're breaking immutability, which we should definitely not do. however, making the objects clones by default would force them into being serializable (i believe, i dont think you can deep copy a function bound to a context) which would mean breaking services.

The way I get around ever dealing with this issue is using object literals for nearly every send, and being very cautious when i dont.

**Comment by tjgionet** [comment]
I completely understand and we can have this conversation this afternoon at the round table.

**Comment by tjgionet** [comment]
So here we will make the object immutable/non-referential to any object defined outside the this.send scope.

**Comment by valyrie97** [comment]
Closed with PR #94
